src/transformers/models/nemotron/modeling_nemotron.py

```diff
@@ -783,8 +783,22 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
 
+        if use_cache and not isinstance(past_key_values, Cache):
+            if past_key_values is None:
+                past_key_values = DynamicCache()
+            else:
+                past_key_values = DynamicCache.from_legacy_cache(past_key_values)
+                logger.warning_once(
+                    "We detected that you are passing `past_key_values` as a tuple of tuples. This is deprecated and "
+                    "will be removed in v4.47. Please convert your cache or use an appropriate `Cache` class "
+                    "(https://huggingface.co/docs/transformers/kv_cache#legacy-cache-format)"
+                )
```

Comment:
I don't think we need to add deprecation message for newly added models, we can support only new Cache objects

Reply:
Thanks for reviewing my code! I removed the deprecation warning. :)

Reply:
sorry if I wasn't clear, I mean totally removing support for the `tuple` format and this the `from_legacy_cache`

I'll ping the core maintainer after that for the final review :)

Reply:
Oh, sorry I got it wrong. Now, I removed support for tuple shaped past_key_values. Is this what you meant?